source/documentation/runbooks/004-anlytical-platform-compute-maintenance.html.md.erb

```diff
@@ -0,0 +1,81 @@
+---
+owner_slack: "#analytical-platform-notifications"
+title: Analytical Platform Compute Maintenance
+last_reviewed_on: 2024-10-28
+review_in: 3 months
+---
+
+# <%= current_page.data.title %>
+
+On the first day of the month the workflow [schedule-issue-compute-infrastructure.yml] (https://github.com/ministryofjustice/analytical-platform/blob/bba07f4908b443a2c15bf94738c8b84c5211e9f9/.github/workflows/schedule-issue-compute-infrastructure.yml) will automatically raise a ticket for [Maintenance - Analytical Platform Compute ] (https://github.com/ministryofjustice/analytical-platform/issues/5661 )
+
+This maintenance ticket includes EKS cluster Upgrade if applicable and/or patching to ensure all components are as up to date as possible.
+
+##  Check for new release
+
+Check if a new release of Amazon EKS Kubernetes has been made available [here](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-release-calendar
+)
+
+If a new EKS release available carry out EKS upgrade and patch all other components
+
+Or
+
+If no EKS release available patch all other components including addons
+
+## The Approach
+
+1. Create a new branch / PR.
+2. Make changes to the code.
+3. Apply them in the command line.
+4. Merge the PR with the changes already applied.
+
+### Order
+
+Apply in the development, test and production environments, resolving any issues before progressing to the next.
+
+## Assumptions
+
+* You are operating in the analytical-platform development container.
+* To interrogate the cluster, you are exec'd into the same account as the cluster you are operating on `aws-sso exec --profile analytical-platform-compute-test:modernisation-platform-developer`
+* Use account `modernisation-platform-developer` for Test and Procuction and `modernisation-platform-sandbox` for Development.
+* If necessary update kubeconfig as follows `aws eks update-kubeconfig --region eu-west-2 --name analytical-platform-compute-test`
+* Set context as follows  `kubectl config use-context arn:aws:eks:eu-west-2:767397661611:cluster/analytical-platform-compute-test`
+* To run terraform locally see the [runbook](https://docs.analytical-platform.service.justice.gov.uk/documentation/runbooks/002-working-in-modernisation-platform.html#working-in-modernisation-platform) for further instructions.
+
```

Comment:
- Elsewhere we use `-` over `*` in the other runbooks so I've altered this to match that.
- Added full stops.
- Altered line 37 for clarity.
- kubeconfig -> `~/kube/config`
```suggestion
- You are operating in the `analytical-platform` repository Development Container.
- To interrogate the cluster, you are exec'd into the same account as the cluster you are operating on `aws-sso exec --profile analytical-platform-compute-test:modernisation-platform-developer`.
- Use account `modernisation-platform-developer` for Test and Production and `modernisation-platform-sandbox` for Development.
- If necessary update `~/kube/config` as follows `aws eks update-kubeconfig --region eu-west-2 --name analytical-platform-compute-test`.
- Set context as follows `kubectl config use-context arn:aws:eks:eu-west-2:767397661611:cluster/analytical-platform-compute-test`.
- To run terraform locally see the [runbook](https://docs.analytical-platform.service.justice.gov.uk/documentation/runbooks/002-working-in-modernisation-platform.html#working-in-modernisation-platform) for further instructions.

```